### PR TITLE
Change SpawELO model to behave more like a regular ML model.

### PR DESCRIPTION
**Pull request by Spawek**
Changed SpawELO model to one using L4 loss, added backtracking.

Also changed training data to use estimated win chance instead of 100%  in each case.

Removed regularization and noise as they are no longer needed.

### CODE REVIEW

File: src-tauri/crates/spawelo/src/lib.rs [hunk 1]
```diff
@@ -78,99 +42,96 @@ pub fn ml_elo(
...
-    elo.into_iter().map(|i| (i.0.clone(), i.1)).collect()
+
+    elo
 }
 
-fn log_elo(elo: &HashMap<&PlayerId, i64>) {
+fn log_elo(elo: &HashMap<PlayerId, f64>) {
     let mut elo: Vec<_> = elo.into_iter().collect();
-    elo.sort_by_key(|p| p.1);
+    elo.sort_by_key(|p| *p.1 as i64 * 1000);
```
Comment (jacekchmiel) [inline src-tauri/crates/spawelo/src/lib.rs]: Isn't it better to multiply before casting to int?

**Review by jacekchmiel — DISMISSED** [review]

File: src-tauri/crates/spawelo/src/lib.rs (hunk 1 above)
Comment (jacekchmiel) [reply src-tauri/crates/spawelo/src/lib.rs]: Multiplying after cast is rather pointless.

File: src-tauri/crates/eloelo_model/src/history.rs [hunk 2]
```diff
@@ -16,6 +16,7 @@ pub struct HistoryEntry {
     pub timestamp: DateTime<Local>,
     pub winner: Vec<PlayerId>,
     pub loser: Vec<PlayerId>,
+    pub win_probability: f64,
```
Comment (jacekchmiel) [inline src-tauri/crates/eloelo_model/src/history.rs:19]: Maybe we shouldn't store that if this is to be derived from win scale and duration?

File: src-tauri/crates/spawelo/src/lib.rs (hunk 1 above)
Comment (Spawek) [reply src-tauri/crates/spawelo/src/lib.rs]: Fixed

**Pull request merged**